packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts

```diff
@@ -77,6 +101,20 @@ export class CodeCommitSourceAction extends Action {
 
     this.branch = branch;
     this.props = props;
+    this._variables = {
+      repositoryName: this.variableExpression('RepositoryName'),
+      branchName: this.variableExpression('BranchName'),
+      authorDate: this.variableExpression('AuthorDate'),
```

Comment:
Why not just instantiate a variables object just-in-time (in `get variables()`)?

Reply:
Yes, to implement https://github.com/aws/aws-cdk/pull/5604#discussion_r365583748 , this is actually require (see my comment: https://github.com/aws/aws-cdk/pull/5604#discussion_r366105877).